plugin/core/views.py

```diff
@@ -802,7 +809,23 @@ def format_completion(
         st_trigger = lsp_filter_text
         if st_details:
             st_details += " | "
-        st_details += "<p>{}</p>".format(html.escape(lsp_label))
+        if lsp_label_details:
+            lsp_label_detail = lsp_label_details.get("detail")
+            lsp_label_description = lsp_label_details.get("description")
+            st_details += _wrap_in_tags(
+                "p",
+                "{label}{detail}{description}".format(
+                    # `label` should be rendered most prominent.
+                    label=_wrap_in_tags("b", lsp_label, escape=True),
+                    # `detail` should be rendered less prominent than `label`.
+                    detail=html.escape(lsp_label_detail) if isinstance(lsp_label_detail, str) else "",
+                    # `description` should be rendered less prominent than `detail`.
+                    description=_wrap_in_tags("i", lsp_label_description, escape=True)
+                ),
```

Comment:
Might be hard to argue about it now, since there are now servers supporting it but the `description` most likely needs to have to be separated with a space at least from the `detail`.

I imagine the possible values could be:

label = "trim"
detail = "(str)"
description = "Trims a string"

in which without a space it would look like:

```
trim(str)Trims a string
```

Should probably use `|` or `-` to separate it.

Reply:
You're probably right. For the `detail` it says there should not be spacing applied. For the `description` it says nothing about space separation. https://github.com/microsoft/language-server-protocol/blob/gh-pages/_specifications/specification-3-17.md#textDocument_completion

Reply:
Yes.
And I see that in tests you have assumed that the field itself would have a space but that's not how I think it would work. The spec itself shouldn't concern itself with presentation (aside from giving suggestions).

Reply:
fixed in a8a4425